api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/ResourcePool.java

```diff
@@ -26,4 +28,9 @@ public Resource getDefault() {
 		return Resource.class;
 	}
 
+	@Override
+	public String getRDFType() {
+		return DYNAPI_RESOURCE;
```

Comment:
The check needs to be against java uri, not ontology class uri (example java:edu.cornell.mannlib.vitro.webapp.dynapi.components.Action)
Can be done with this method
https://github.com/vivo-project/Vitro/blob/39e1eed20df592e8c0b62834d34042bc4d370ae3/api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java#L34

Reply:
Not working.

```
private boolean isInModel(String uri) {
        Resource s = dynamicAPIModel.getResource(uri);
        Property p = dynamicAPIModel.getProperty(RDF_TYPE); 
        ObjectProperty o = dynamicAPIModel.getObjectProperty(toJavaUri());

        StmtIterator it = dynamicAPIModel.listStatements();
        while(it.hasNext()) {
            Statement st = it.next();
            if (st.toString().contains(uri) &&  st.toString().contains(toJavaUri())) {
                System.out.println("Found: ");
                System.out.println(st);
            }
        }

        System.out.println("\n" + toJavaUri() + " => " + dynamicAPIModel.getObjectProperty(toJavaUri()) + "\n");
        System.out.println("\n" + ConfigurationBeanLoader.toJavaUri(getType()) + " => " + dynamicAPIModel.getObjectProperty(ConfigurationBeanLoader.toJavaUri(getType())) + "\n");

        System.out.println("\n" + s + ", " + p + ", " + toJavaUri() + " => " + dynamicAPIModel.contains(s, p, toJavaUri()) + "\n");
        System.out.println("\n" + s + ", " + p + ", " + ConfigurationBeanLoader.toJavaUri(getType()) + " => " + dynamicAPIModel.contains(s, p, ConfigurationBeanLoader.toJavaUri(getType())) + "\n");

        return dynamicAPIModel.contains(s, p, toJavaUri());
    }

    private String toJavaUri() {
        Class<C> clazz = getType();
        String javaUri = ConfigurationBeanLoader.toJavaUri(clazz);
        String simpleName = clazz.getSimpleName();

        return javaUri.replace(format(".%s", simpleName), format("#%s", simpleName));
    }
 ```
 
 ```
 Found:
[https://vivoweb.org/ontology/vitro-dynamic-api/action/testReloadAction1, http://www.w3.org/1999/02/22-rdf-syntax-ns#type, java:edu.cornell.mannlib.vitro.webapp.dynapi.components#Action]

java:edu.cornell.mannlib.vitro.webapp.dynapi.components#Action => null


java:edu.cornell.mannlib.vitro.webapp.dynapi.components.Action => null


https://vivoweb.org/ontology/vitro-dynamic-api/action/testReloadAction1, http://www.w3.org/1999/02/22-rdf-syntax-ns#type, java:edu.cornell.mannlib.vitro.webapp.dynapi.components#Action => false


https://vivoweb.org/ontology/vitro-dynamic-api/action/testReloadAction1, http://www.w3.org/1999/02/22-rdf-syntax-ns#type, java:edu.cornell.mannlib.vitro.webapp.dynapi.components.Action => false
```

Reply:
@litvinovg thoughts?

Reply:
@litvinovg I have implemented by listing object properties for the resource and property and checking if any match the expected Java URI.